taskw/fields/depends.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+from distutils.version import LooseVersion
+
+import re
+import uuid
+
+from .base import DirtyableList, Field
+
+
+class DependsField(Field):
+    def deserialize(self, value):
+        if not value:
+            return DirtyableList([])
+
+        try:
+            # In task-2.5, this moved from a comma-separated string to a real list.
+            # here we allow a migration path where old splitable strings are
+            # handled as well as newschool lists.
+            if hasattr(value, "split"):
+                value = value.split(",")
+
+            return DirtyableList([uuid.UUID(v) for v in value])
+        except ValueError:
+            dependency_finder = re.compile(
+                "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+            )
+            return DirtyableList(
+                [uuid.UUID(v) for v in dependency_finder.findall(value)]
+            )
+
+    def serialize(self, value):
+        if not value:
+            value = []
+
+        if not hasattr(value, "__iter__"):
+            raise ValueError("Value must be list or tuple, not %r." % value)
+
+        if self.version < LooseVersion("2.5"):
+            return ",".join([str(v) for v in value])
+        else:
+            # We never hit this second code branch now.  taskwarrior changed
+            # API slightly in version 2.5, but we're just going to go with
```

Comment:
Also don't think this is true, either!  See https://taskwarrior.org/docs/design/task.html#attr_depends.